src/ncp/ncp_base.cpp

```diff
@@ -1978,7 +1978,11 @@ template <> otError NcpBase::HandlePropertyGet<SPINEL_PROP_CAPS>(void)
 
 template <> otError NcpBase::HandlePropertyGet<SPINEL_PROP_NCP_VERSION>(void)
 {
+#if OPENTHREAD_RADIO
+    return mEncoder.WriteUtf8(otGetRadioVersionString(mInstance));
```

Comment:
I suggest keeping the mapping between APIs and spinel properties simple and staright-forward and not tied to other configs:
- `otGetVersionString()` maps to `SPINEL_PROP_NCP_VERSION` and gives version of OT stack
- `otGetRadioVersionString()`  maps to `SPINEL_PROP_RCP_VERSION` and gives version of radio platform stack (RCP version in the case the OT stack is being used in host-RCP architecture).

----

I think the idea behind `otGetRadioVersionString()` API was to provide the RCP version when OT stack is being used in host-RCP architecture (OT stack is being built as host of an RCP).

If we have `OPENTHREAD_RARDIO` config, it indicates OT is being built as RCP firmware itself, then I guess it does not make sense for RCP firmware to use host/RCP model and have its own RCP (so `otGetRadioVersionString()` may not be applicable under this).

----

If we want to provide more information in the `otGetVersionString()` about the platform (e.g. radio specific info/version, etc), I think it can be added as part of `OPENTHREAD_CONFIG_PLATFORM_INFO` config (which will be part of the OT version string):
https://github.com/openthread/openthread/blob/018f54f71b059bf338247595eb1f6a6710e3d0c9/src/core/api/instance_api.cpp#L178-L189

Reply:
@abtink thanks for reviewing. I understand what you mean. In our case, for instance, we can't use `OPENTHREAD_CONFIG_PLATFORM_INFO` because we need to build the version in runtime.

I can close the PR and keep this change private in our branch.

Reply:
In any case, do you think the `rcp version` cli command is doing the right thing, returning `otGetVersionString` instead of `otGetRadioVersionString`?:

https://github.com/openthread/openthread/blob/3350f598f6759fd7de0c80845a925197458d0420/src/cli/cli.cpp#L3608-L3617

Reply:
I think it is outputting `otPlatRadioGetVersionString()`, right? Maybe I am missing something?

I see the it is checking it against `otGetVersionString()` and returns `OT_ERROR_NOT_IMPLEMENTED` if it matches  
- It is pointer check as well, so only if they are both pointing to same string buffer :)
- I guess the code is indirectly trying  to see if `otPlatRadioGetVersionString()` is provided or the weak version of it is being used.